### ISSUE TITLE
Scratch Org Created Users No Longer Displayed via force:org:list

### ISSUE THREAD

**Issue by aesanch2** [forcedotcom/cli#951]
### Summary

Since updating to the latest sfdx cli, we've encountered issues with users that are created via the `sfdx force:user:create` command no longer being displayed in the `force:org:list`. As our pipeline requires the creation of users for testing and then using their connection information to update configurations, we are no longer able to build development environments for our project.

### Steps To Reproduce:

**Repository to reproduce:** [dreamhouse-lwc](https://github.com/dreamhouseapp/dreamhouse-lwc)

1.  Create a scratch org via `sfdx force:org:create -s -f config/project-scratch-def.json -a dreamhouse`.
2. Create a new scratch org user via `sfdx force:user:create username=testuser1@my.org email=me@my.org`.
3. List the current orgs via `sfdx force:org:list`.

### Expected result

All scratch org connections and any scratch org users should be displayed.
```
     ALIAS                        USERNAME                       ORG ID              EXPIRATION DATE
───  ───────────────────────────  ─────────────────────────────  ──────────────────  ───────────────
(U)  dreamhouse                   test-rzt0hdiza30k@example.com  00D2F000000GdHPUA0  2021-04-06
                                  testuser1@my.org               00D2F000000GdHPUA0  2021-04-06
```

### Actual result

Only primary scratch orgs are displayed.
```
     ALIAS                        USERNAME                       ORG ID              EXPIRATION DATE
───  ───────────────────────────  ─────────────────────────────  ──────────────────  ───────────────
(U)  dreamhouse                   test-rzt0hdiza30k@example.com  00D2F000000GdHPUA0  2021-04-06
```

### Additional information

**SFDX CLI Version**: sfdx-cli/7.93.1-762bca056d darwin-x64 node-v14.15.4

**SFDX plugin Version**: 
```
@oclif/plugin-autocomplete 0.3.0 (core)
@oclif/plugin-commands 1.3.0 (core)
@oclif/plugin-not-found 1.2.4 (core)
@oclif/plugin-plugins 1.10.0 (core)
@oclif/plugin-warn-if-update-available 1.7.0 (core)
@oclif/plugin-which 1.0.3 (core)
@salesforce/lwc-dev-server 2.10.0
├─ @oclif/plugin-update 1.3.10
└─ @oclif/plugin-help 2.2.3
@salesforce/sfdx-trust 3.6.0 (core)
alias 1.1.8 (core)
auth 1.5.1 (core)
config 1.2.7 (core)
generator 1.1.5 (core)
org 1.6.0
salesforcedx 51.5.0
├─ schema 1.0.4
├─ limits 1.0.4
├─ user 1.2.0
├─ custom-metadata 1.0.11
├─ apex 0.1.4
├─ @salesforce/sfdx-plugin-lwc-test 0.1.7
├─ templates 51.3.0
└─ salesforce-alm 51.6.3
sfdx-cli 7.93.1 (core)
telemetry 1.1.1 (core)
```

**OS and version**: macOS 10.15.7

**Comment by micha79x** [comment]
Might be related to https://github.com/forcedotcom/cli/issues/940?

**Comment by mshanemc** [comment]
yes, duplicate of #940 , which has a fix you can use now.

**Comment by aesanch2** [comment]
@mshanemc I updated the org plugin as suggested in the referenced issue and I'm still not seeing Scratch Org users in the display response.

```
❯ sfdx plugins --core
@oclif/plugin-autocomplete 0.3.0 (core)
@oclif/plugin-commands 1.3.0 (core)
@oclif/plugin-not-found 1.2.4 (core)
@oclif/plugin-plugins 1.10.0 (core)
@oclif/plugin-warn-if-update-available 1.7.0 (core)
@oclif/plugin-which 1.0.3 (core)
@salesforce/lwc-dev-server 2.10.0
├─ @oclif/plugin-help 2.2.3
└─ @oclif/plugin-update 1.3.10
@salesforce/sfdx-trust 3.6.0 (core)
alias 1.1.8 (core)
auth 1.5.1 (core)
config 1.2.7 (core)
generator 1.1.5 (core)
org 1.6.1
salesforcedx 51.5.0
├─ limits 1.0.4
├─ schema 1.0.4
├─ apex 0.1.4
├─ custom-metadata 1.0.11
├─ templates 51.3.0
├─ user 1.2.0
├─ @salesforce/sfdx-plugin-lwc-test 0.1.7
└─ salesforce-alm 51.6.3
sfdx-cli 7.93.1 (core)

```

**Comment by nigelhughesfairsailcom** [comment]
This is still happening, following sfdx force:user:create the org goes missing.  

```
λ sfdx plugins --core
@oclif/plugin-autocomplete 0.3.0 (core)
@oclif/plugin-commands 1.3.0 (core)
@oclif/plugin-help 3.3.1 (core)
@oclif/plugin-not-found 1.2.6 (core)
@oclif/plugin-plugins 1.10.11 (core)
@oclif/plugin-update 1.5.0 (core)
@oclif/plugin-warn-if-update-available 1.7.3 (core)
@oclif/plugin-which 1.0.4 (core)
@salesforce/analytics 1.0.9
@salesforce/sfdx-plugin-lwc-test 1.0.0 (core)
@salesforce/sfdx-scanner 2.13.7
alias 2.1.0 (core)
apex 1.1.0 (core)
auth 2.2.3 (core)
community 2.0.0 (core)
config 1.4.14 (core)
custom-metadata 2.0.0 (core)
data 2.0.4 (core)
generator 2.0.2 (core)
info 2.0.1 (core)
isvte-sfdx-plugin 1.1.16
limits 2.0.1 (core)
org 2.0.5 (core)
salesforce-alm 54.6.3 (core)
schema 2.1.1 (core)
sfdmu 4.15.0
sfdx-cli 7.161.0 (core)
signups 1.2.0 (core)
source 2.0.9 (core)
telemetry 2.0.0 (core)
templates 55.0.0 (core)
trust 2.0.1 (core)
user 2.1.0 (core)
```

**Comment by peternhale** [comment]
@nigelhughesfairsailcom would you post the results of `sfdx version --verbose` please.
